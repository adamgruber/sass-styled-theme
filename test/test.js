const path = require('path');
const sassToStyledTheme = require('../index');

const testfiles = {
  basic: path.resolve(__dirname, './sass/test-basic.scss'),
  camel: path.resolve(__dirname, './sass/test-opts-camel.scss'),
  sassOpts: path.resolve(__dirname, './sass/test-opts-sass.scss'),
  import: path.resolve(__dirname, './sass/test-import.scss'),
};

describe('sassToStyledTheme', () => {
  it('should convert basic SASS vars', () => {
    expect(sassToStyledTheme(testfiles.basic)).toMatchSnapshot();
  });

  it('should convert keys to camelCase', () => {
    expect(sassToStyledTheme(testfiles.camel, { camel: true })).toMatchSnapshot();
  });

  it('should handle imports', () => {
    expect(sassToStyledTheme(testfiles.import)).toMatchSnapshot();
  });

  it('should pass sass options through', () => {
    const opts = {
      sassOptions: {
        includePaths: [path.join(__dirname, 'sass', 'nested')],
      },
    };
    expect(sassToStyledTheme(testfiles.sassOpts, opts)).toMatchSnapshot();
  });
});
