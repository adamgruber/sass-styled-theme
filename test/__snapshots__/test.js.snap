// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sassToStyledTheme should convert basic SASS vars 1`] = `
Object {
  "boolean": true,
  "builtinFunc": "rgb(255, 51, 51)",
  "calculated": 200,
  "customFunc": "rgb(255, 0, 0)",
  "hexColor": "rgb(240, 240, 240)",
  "importantStyle": "10px !important",
  "inline": 100,
  "mixedList": "1px solid rgb(0, 0, 0)",
  "namedColor": "rgb(255, 0, 0)",
  "nestedGlobal": "100px",
  "null": null,
  "number": 100,
  "numberList": "0 10px 1em 10%",
  "numberWithUnit": "100px",
  "quotedString": "quoted",
  "rgbColor": "rgb(255, 255, 255)",
  "rgbaColor": "rgba(255, 255, 255, 0.5)",
  "stringList": "'Helvetica', 'Arial', sans-serif",
  "unquotedString": "string",
  "valueMap": Object {
    "key-two": "key",
    "keyThree": false,
    "key_one": "2em",
  },
}
`;

exports[`sassToStyledTheme should convert keys to camelCase 1`] = `
Object {
  "myCoolVar": "400px",
  "myOtherVAr": "300px",
  "myVar": "500px",
  "myVar2": "600px",
  "myVar3": "700px",
}
`;

exports[`sassToStyledTheme should handle imports 1`] = `
Object {
  "importedVar": true,
  "myVar": "150px",
}
`;

exports[`sassToStyledTheme should pass sass options through 1`] = `
Object {
  "myVar": "100px",
  "nestedVar": false,
}
`;
